### PR TITLE
[Minor] Move combine_join util to under equivalence.rs

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

## Rationale for this change
Utils to construct `join_equivalence` and `join_ordering_equivalence` are move under `equivalence.rs` file. Computations regarding the equivalence and ordering equivalence are gathered in the `equivalence.rs` file. With this change code organization is bit better.

### CODE REVIEW

File: datafusion/physical-expr/src/equivalence.rs [hunk 1]
```diff
@@ -885,6 +885,240 @@ fn req_satisfied(given: LexOrderingRef, req: &[PhysicalSortRequirement]) -> bool
     true
 }
 
+/// Combine equivalence properties of the given join inputs.
+pub fn combine_join_equivalence_properties(
```
Comment (mustafasrepo) [inline datafusion/physical-expr/src/equivalence.rs:889]: This function is cop-pasted

File: datafusion/physical-expr/src/equivalence.rs [hunk 2]
```diff
@@ -885,6 +885,240 @@ fn req_satisfied(given: LexOrderingRef, req: &[PhysicalSortRequirement]) -> bool
...
+            let new_column2 =
+                Column::new(column2.name(), left_columns_len + column2.index());
+            new_properties.add_equal_conditions((column1, &new_column2))
+        })
+    }
+    new_properties
+}
+
+/// Calculate equivalence properties for the given cross join operation.
+pub fn cross_join_equivalence_properties(
```
Comment (mustafasrepo) [inline datafusion/physical-expr/src/equivalence.rs:941]: This function is copy-pasted

File: datafusion/physical-expr/src/equivalence.rs [hunk 3]
```diff
@@ -885,6 +885,240 @@ fn req_satisfied(given: LexOrderingRef, req: &[PhysicalSortRequirement]) -> bool
...
+/// join schema consists of a combination of left and right schema (Inner,
+/// Left, Full, Right joins). Then, we normalize the sort expressions of
+/// ordering equivalences one by one. We make sure that each expression in the
+/// ordering equivalence is either:
+/// - The head of the one of the equivalent classes, or
+/// - Doesn't have an equivalent column.
+///
+/// This way; once we normalize an expression according to equivalence properties,
+/// it can thereafter safely be used for ordering equivalence normalization.
+fn get_updated_right_ordering_equivalent_class(
```
Comment (mustafasrepo) [inline datafusion/physical-expr/src/equivalence.rs:981]: This function is copy pasted

File: datafusion/physical-expr/src/equivalence.rs [hunk 4]
```diff
@@ -885,6 +885,240 @@ fn req_satisfied(given: LexOrderingRef, req: &[PhysicalSortRequirement]) -> bool
...
+                let mut orderings = vec![];
+                for left_ordering in left_oeq_class.iter() {
+                    for right_ordering in updated_right_oeq.iter() {
+                        let mut ordering = left_ordering.to_vec();
+                        ordering.extend(right_ordering.to_vec());
+                        let ordering_normalized =
+                            join_eq_properties.normalize_sort_exprs(&ordering);
+                        orderings.push(ordering_normalized);
+                    }
+                }
```
Comment (mustafasrepo) [inline datafusion/physical-expr/src/equivalence.rs:1056]: Previously we were prefixing with left ordering. We now prefix with all of the orderings inside left equivalence. We no longer use left output_ordering. This is the only change. in this function.

File: datafusion/physical-expr/src/equivalence.rs [hunk 5]
```diff
@@ -885,6 +885,240 @@ fn req_satisfied(given: LexOrderingRef, req: &[PhysicalSortRequirement]) -> bool
...
+                let mut orderings = vec![];
+                for right_ordering in right_oeg_class.iter() {
+                    for left_ordering in left_oeq_class.iter() {
+                        let mut ordering = right_ordering.to_vec();
+                        ordering.extend(left_ordering.to_vec());
+                        let ordering_normalized =
+                            join_eq_properties.normalize_sort_exprs(&ordering);
+                        orderings.push(ordering_normalized);
+                    }
+                }
```
Comment (mustafasrepo) [inline datafusion/physical-expr/src/equivalence.rs:1107]: Similar to above comment. However, in this case left orderings are prefixed with right orderings

File: datafusion/physical-expr/src/equivalence.rs [hunk 6]
```diff
@@ -1131,4 +1365,82 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_get_updated_right_ordering_equivalence_properties() -> Result<()> {
```
Comment (mustafasrepo) [inline datafusion/physical-expr/src/equivalence.rs:1370]: This test is copy-pasted

**Review by alamb — APPROVED** [review]
looks like a nice change to me -- thank you @mustafasrepo

**Pull request merged**